### PR TITLE
修正map没有初始化而直接使用时报assignment to entry in nil map错误的问题

### PR DESCRIPTION
**Pull request by tzwsoho**

### CODE REVIEW

File: tars/tools/tars2go/gen_go.go [hunk 1]
```diff
@@ -1414,6 +1414,11 @@ func (gen *GenGo) genSwitchCase(tname string, fun *FunInfo) {
 	outArgsCount := 0
 	for _, v := range fun.Args {
 		c.WriteString("var " + v.Name + " " + gen.genType(v.Type) + "\n")
+		if v.Type.Type == tkTMap {
+			tk := gen.genType(v.Type.TypeK)
+			tv := gen.genType(v.Type.TypeV)
+			c.WriteString(v.Name + " = make(map[" + tk + "]" + tv + ")\n")
```
Comment (chenhengqi) [inline tars/tools/tars2go/gen_go.go:1420]: How about `c.WriteString(v.Name + " = make(" + gen.genType(v.Type) + ")\n")`

**Comment by chenhengqi** [comment]
It looks good.

Again you should squash your commits and force push. FYI see this [link](https://www.internalpointers.com/post/squash-commits-into-one-git).

**Comment by tzwsoho** [comment]
Hi, I squashed my commits. Please check, thanks!

**Comment by chenhengqi** [comment]
Ah, it seems your commit contains other contributor's works.
You should keep your fork [up to date](https://stefanbauer.me/articles/how-to-keep-your-git-fork-up-to-date).

**Comment by tzwsoho** [comment]
I think it's ok now. Please check again, thank you very much.

**Comment by chenhengqi** [comment]
Everything is fine except commit message, could you amend that ?

**Review by chenhengqi — APPROVED** [review]

**Comment by tzwsoho** [comment]
> Everything is fine except commit message, could you amend that ?

Final version, hope ok now. Please check again, thanks a lots!!!

**Pull request merged**